src/adapters/fsFileSystem.ts

```diff
@@ -12,7 +13,8 @@ export const fsFileSystem: FileSystem = {
     },
     readFile: async (filePath: string) => {
         try {
-            return (await fs.promises.readFile(filePath)).toString();
+            const readFile = promisify(fs.readFile);
```

Comment:
Nit: would you mind moving this variable to the top of the file? That way we don't `promisify` each time this gets called.